src/testdrive/src/action/kafka/ingest.rs

```diff
@@ -98,6 +106,15 @@ impl Transcoder {
                         out.write_i32::<NetworkEndian>(*schema_id).unwrap();
                     }
                     out.extend(avro::to_avro_datum(&schema, val).map_err(|e| e.to_string())?);
+                    if corrupt_buffer {
```

Comment:
It seems better to move these `corrupt_buffer` sections closer to where they're invoked, in the `impl Action`.

Otherwise it's a bit confusing what this `transcode` block does and why it sometimes needs to corrupt the buffer.

Also, FWIW: for avro here you only need to corrupt the first byte. That should already cause decoding to fail, because Confluent avro format starts with a magic `\0` byte.

Reply:
I feel like the transcode block needs to corrupt the buffer because it has the context on the encoding type.

Happy to only corrupt the first byte for Avro, but I'll need to make sure that this still causes the `confluent_wire_format = false` testcase to fail.

Reply:
ok, makes sense